eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -1532,6 +1532,17 @@ default T detectIfNone(Predicate<? super T> predicate, Function0<? extends T> fu
             Function<? super T, ? extends NK> keyFunction,
             Function<? super T, ? extends NV> valueFunction);
 
+    /**
+     * Converts the collection to a MutableSortedMap implementation using the specified key and value functions
+     * and sorts it based on the natural order of the attribute returned by {@code sortBy} function.
+     *
+     * @since 10.0
+     */
+    <KK extends Comparable<? super KK>, NK, NV> MutableSortedMap<NK, NV> toSortedMapBy(
```

Comment:
We can merge this in a minor version as well. If you convert this to a default method and throw.

Reply:
Refer to org.eclipse.collections.api.factory.list.MutableListFactory#withInitialCapacity for example.

Reply:
Great idea! And what version should be specified in `@since`?

Reply:
The next minor release will be 9.2.

Reply:
IMO you could take out @since for now,. 

We could add it when we start merging 10.0 changes, altogether with other default methods that currently throw and start to be enabled from 10.0.

Reply:
Added default method with UnsupportedOperationException and removed `@since`